sqls/activity-repo-top-Chinese/sql

```diff
@@ -0,0 +1,35 @@
+    activity_list.commits AS commits,activity_list.additions AS additions,activity_list.deletions AS deletions
+    FROM
+        (SELECT contribute_list.repo_id AS repo_id, round(sum(sqrt(contribute_list.score)),2) AS repo_activity,
+        COUNTDistinct(contribute_list.actor_id) AS actor_count,  SUM(icc.count) as icc_count,SUM(oic.count) as oic_count, SUM(opc.count) as opc_count, SUM(rcc.count) as rcc_count, SUM(mpc.count) as mpc_count,
+        SUM(contribute_list.commits) AS commits,SUM(contribute_list.additions) AS additions,SUM(contribute_list.deletions) AS deletions
+        FROM
+        (SELECT
+        icc.repo_id AS repo_id, icc.actor_id AS actor_id, icc.count, oic.count, opc.count, rcc.count, mpc.count,
+        {{weight.issueCommentWeight}}*icc.count+{{weight.openIssueWeight}}*oic.count+{{weight.openPullWeight}}*opc.count+{{weight.pullReviewWeight}}*rcc.count+{{weight.mergePullWeight}}*mpc.count AS score,
+        mpc.commits AS commits,mpc.additions AS additions,mpc.deletions AS deletions
+        FROM
+            (SELECT repo_id, actor_id, COUNT(*) count FROM {{table}} WHERE type='IssueCommentEvent' AND action='created' GROUP BY repo_id, actor_id HAVING repo_id IN {{[...repos.values()]}}) AS icc
+            LEFT JOIN
+            (SELECT repo_id, actor_id, COUNT(*) count FROM {{table}} WHERE type='IssuesEvent' AND action='opened' GROUP BY repo_id, actor_id HAVING repo_id IN {{[...repos.values()]}}) AS oic
+            ON icc.repo_id=oic.repo_id AND icc.actor_id=oic.actor_id
+            LEFT JOIN
+            (SELECT repo_id, actor_id, COUNT(*) count FROM {{table}} WHERE type='PullRequestEvent' AND action='opened' GROUP BY repo_id, actor_id HAVING repo_id IN {{[...repos.values()]}}) AS opc
+            ON icc.repo_id=opc.repo_id AND icc.actor_id=opc.actor_id
+            LEFT JOIN
+            (SELECT repo_id, actor_id, COUNT(*) count FROM {{table}} WHERE type='PullRequestReviewCommentEvent' AND action='created' GROUP BY repo_id, actor_id HAVING repo_id IN {{[...repos.values()]}}) AS rcc
+            ON icc.repo_id=rcc.repo_id AND icc.actor_id=rcc.actor_id
+            LEFT JOIN
+            (SELECT repo_id, issue_author_id AS actor_id, COUNT(*) as count,SUM(pull_commits) AS commits,SUM(pull_additions) AS additions,SUM(pull_deletions) AS deletions FROM {{table}} WHERE type='PullRequestEvent' AND action='closed' AND pull_merged=1 GROUP BY repo_id, actor_id HAVING repo_id IN {{[...repos.values()]}}) AS mpc
+            ON icc.repo_id=mpc.repo_id AND icc.actor_id=mpc.actor_id
+        ) AS contribute_list
+        GROUP BY repo_id
+        ORDER BY repo_activity DESC
+        LIMIT {{topN}}) activity_list
+        INNER JOIN
+        (SELECT repo_id,anyHeavy(repo_name) AS repo_name,anyHeavy(repo_language) AS repo_language FROM
```

Comment:
Language stats should be like global SQL

Reply:
I didn't realize that in the global rank list SQL, the language query still is another join table. Actually we can use `PullRequestEvent` to fetch the language data because language field only exists in `PullRequestEvent`.

Reply:
Many PRs language is null. If I use same conditions to filter PRs, the number of PRs is not right.

Reply:
You can use `max` instead of `anyHeavy` to filter the empty values since language is a quite stable column.

Reply:
Can I use max function in string field? Anyway, one repo may have some languages. So I use anyHeavy function to get more frequent language in repo.

Reply:
Yes, you can use `max` on string field. Actually in the log data, only dominate language is included and for most repos(especially most active repos), dominate language does not change. So maybe you can use `max` in first level aggregation and `anyHeavy` for second level aggregation if you concern about language change.

Reply:
Most PRs language is null. If I don’t judge null condition, the anyHeavy function will return null.

Reply:
I agree that for most situations, using max function is enough. I will fix it.

Reply:
I think it's fine to collect repo name and language in this way, we need to figure out why so many PR events are without language field.